### PR TITLE
Hotfix/sqs delete message batch

### PR DESCRIPTION
**Pull request by jeskew**
Addresses issue raised in #637 and #630.

### CODE REVIEW

**Comment by jeremeamia** [comment]
Looks good!

File: tests/Integ/SqsBatchSerializationTest.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                    return [
+                        'Id' => $message['MessageId'],
+                        'ReceiptHandle' => $message['ReceiptHandle'],
+                    ];
+                },
+                $messages->get('Messages')
+            )
+        ]);
+    }
+
```
Comment (jeremeamia) [inline tests/Integ/SqsBatchSerializationTest.php]: Whitespace

File: tests/Api/Serializer/QuerySerializerTest.php [hunk 2]
```diff
@@ -46,4 +46,45 @@ function () {}
         $this->assertEquals('http://foo.com', (string) $request->getUri());
         $this->assertEquals('Action=foo&Version=1&baz=', (string) $request->getBody());
     }
+
+    public function testUsesLocationNameToIdentifyMembersOfFlattenedLists()
```
Comment (jeremeamia) [inline tests/Api/Serializer/QuerySerializerTest.php]: Are you going to replace this with the protocol test?

**Comment by jeremeamia** [comment]
I'm going to merge this so the fix is applied, but we can update the test later.

**Comment by docteurklein** [comment]
still have some weird errors about batch delete:
```

**Pull request merged**